### PR TITLE
refactored

### PR DESCRIPTION
**Pull request by st-gb**
Hi Thomas.

Endlich habe ich Zeit und Muße gehabt, das umzusetzen (Deine E-Mail vom 04.Juli soll nicht umsinst gewesen sein ;-) ).
Gucke Dir mal die Änderungen an und übernimm sie dann.

Gru

### CODE REVIEW

**Comment by t-b** [comment]
@st-gb Sorry for the late response.

I like the per check error messages but I don't think that creating a separate file for the test function itself is good.

I'm replying with some more comments inline.

Please also:
- Explain why you did the change in the commit message
- Don't commit commented out code

File: Readme.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## Motivation
+Avoid git errors like "git checkout failed" for file names when pulling from a MS Windows client.
+File name that are invalid for MS Windows may have been check in under Linux.
```
Comment (t-b) [inline Readme.md]: "File name" -> "File names"
"check in" -> "checked in"

File: Readme.md [hunk 2]
```diff
@@ -9,6 +13,7 @@ you can then enable the hook.
 
 ## Testing
 Clone the repository and execute ./test-hook.sh
+Also testIfWinValidFileName.sh may be tested in advance to git commit to check for Windows filename compatibility to avoid too much I/O.
```
Comment (t-b) [inline Readme.md]: Can be removed as `testIfWinValidFileName.sh` will not be kept.

File: testIfWinValidFileName.sh [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+   then
+    cause="file name all periods"
+  fi 
+  if test $absolutepathtoolong -ne 0
+   then
+    cause="absolute path too long"
+  fi
+  if test "$cause" != ""
+  then
+  #else
```
Comment (t-b) [inline testIfWinValidFileName.sh:79]: It would be nice to support multiple causes per file in the error message.

**Review by t-b — CHANGES_REQUESTED** [review]

**Comment by t-b** [comment]
@st-gb Please write in english so that others can follow as well.

File: Readme.md (hunk 2 above)
Comment (st-gb) [reply Readme.md]: Why shouldn't the file "testIfWinValidFileName.sh" be kept?
I think it is senseful to use a function that can be used by both the hook itself and to test externally.
Would be using a function "testIfWinValidFileName" inside an existing shell file be ok?

File: testIfWinValidFileName.sh (hunk 3 above)
Comment (st-gb) [reply testIfWinValidFileName.sh:79]: Yes, I can do this.

File: Readme.md (hunk 1 above)
Comment (st-gb) [reply Readme.md]: Also "a MS" -> "an MS" because it depends on the pronunciation ?

**Comment by st-gb** [comment]
Hi Thomas.
So I have to update my branch with a new commit including your requested changes, right?

File: Readme.md (hunk 2 above)
Comment (t-b) [reply Readme.md]: Having two files makes it harder for users, and I think it should be easy to use.

**Comment by t-b** [comment]
@st-gb Yes please add a new commit or even better squash your new changes into the existing commit and force push.

**Comment by st-gb** [comment]
I wrote a concatenate function for the causes.
But I would like the testing be callable also from outside the pre-commit hook.

**Comment by st-gb** [comment]
You can see the latest changes here:
https://github.com/st-gb/git-pre-commit-hook-windows-filenames/tree/suggestion_by_StGb

Now we have to decide whether single shell file for pre-commit file or not.
pro & contra:
pro: test without changing the repository.
contra: current path problems (when called from another directory than where the pre-commit resides maybe  testIfWinValidFileName.sh can't be found)

**Comment by t-b** [comment]
@st-gb Sorry for not getting back earlier. As mentioned I'd like to stick with single file as it is more convenient.

**Comment by st-gb** [comment]
I strongly prefer my solution with a dedicated shell script for both testing and changing because of "separation of concerns" principle.
If you stick to your suggestion then I'd like to put code from my "testIfWinValidFileName.sh" into function(s) within the "pre-commit" shell script. Please do this in your repository at first.
I need some more time to do this for my one because I am busy ATM.

**Comment by t-b** [comment]
No reply for two years closing.

**Pull request closed without merging**